File: internal/models/receipt.go
package models

import "time"

type ReceiptCreate struct {
	EmployeeId *string
	CardNumber *string
	PrintDate  *time.Time
	TotalSum   *float64
	VAT        *float64
}

type ReceiptRetrieve struct {
	ReceiptNumber *string
	EmployeeId    *string
	CardNumber    *string
	PrintDate     *time.Time
	TotalSum      *float64
	VAT           *float64
}

type ReceiptUpdate struct {
	EmployeeId *string
	CardNumber *string
	PrintDate  *time.Time
	TotalSum   *float64
	VAT        *float64
}

type ReceiptCreateComplete struct {
	EmployeeId *string
	CardNumber *string
	PrintDate  *time.Time
	Items      []ReceiptItem
}

type ReceiptItem struct {
	UPC           *string
	ProductNumber *int
	SellingPrice  *float64
}

File: frontend/src/types/receipt.ts
export interface ReceiptItem {
  upc: string;
  product_number: number;
  selling_price: number;
}

export interface ReceiptCreate {
  employee_id: string;
  card_number?: string;
  print_date: string;
  sum_total: number;
  vat: number;
}

export interface ReceiptCreateComplete {
  employee_id: string;
  card_number?: string | null;
  print_date: string;
  items: ReceiptItem[];
}

export interface ReceiptRetrieve {
  receipt_number: string;
  employee_id: string;
  card_number?: string | null;
  print_date: string;
  sum_total: number;
  vat: number;
}

export interface ReceiptUpdate {
  employee_id?: string;
  card_number?: string;
  print_date?: string;
  sum_total?: number;
  vat?: number;
}

export interface ReceiptCreateResponse {
  receipt_number: string;
  print_date: string;
  total_sum: number;
  vat: number;
}
